projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -475,12 +474,23 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
-    private setUpCloseOnEscape(info: OverlayInfo) {
-        const wrapperElement = info.elementRef.nativeElement.parentElement.parentElement;
-        fromEvent(wrapperElement, 'keydown').pipe(
+    private setUpCloseOnEscape() {
+        if (this._overlayInfos.length - this._overlayInfos.filter(x => x.closeAnimationPlayer
+            && x.closeAnimationPlayer.hasStarted()).length === 1) {
+            this.subscribeOnEscape(this._document);
+        }
+    }
+
+    private subscribeOnEscape(target: FromEventTarget<unknown>) {
```

Comment:
we do not need this method anymore. Move this logic in `setUpCloseOnEscape` (or `addCloseOnEscapeListener`)